app/components/contacts/CreateNewContact.tsx

```diff
@@ -103,28 +105,30 @@ const CreateNewContact = ({
     target?.select();
   };
 
-  const validate = () => {
+  const validate = (): ValidateError | null => {
     const nicknameRegex = /^([a-zA-Z0-9_-])$/;
     if (nicknameRegex.test(nickname)) {
-      return 'Nickname is missing or invalid';
+      return { type: 'name', message: 'Nickname is missing or invalid' };
+    }
+    if (contacts.some((contact) => contact.nickname === nickname)) {
+      return { type: 'name', message: 'Nickname should be unique' };
     }
     if (!validateAddress(address)) {
-      return 'Address is invalid';
+      return { type: 'address', message: 'Address is invalid' };
     }
-    let retVal = '';
-    contacts.forEach((contact: Contact) => {
-      if (contact.nickname === nickname) {
-        retVal = 'Nickname should be unique';
-      }
-    });
-    return retVal;
+    if (contacts.some((contact) => contact.address === address)) {
+      return {
+        type: 'address',
+        message: 'Contact with the same Account Address already exists',
```

Comment:
Let's help to figure out which one

```suggestion
        message: `Contact ${contact.nickname} has the same Address`,
```